go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		return oc.handlePeerPodSelectorDelete(extraParameters.gp, obj)
+
+	case factory.PeerNamespaceSelectorTypeVar:
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		namespace := obj.(*kapi.Namespace)
+		// Remove namespace address set from the *gress policy in cache
+		// (done in gress.delNamespaceAddressSet()), and then update ACLs
+		return oc.handlePeerNamespaceSelectorOnUpdate(extraParameters.np, extraParameters.gp, func() bool {
+			// ... on condition that the removed address set was in the 'gress policy
+			return extraParameters.gp.delNamespaceAddressSet(namespace.Name)
+		})
+
+	case factory.PeerPodForNamespaceAndPodSelectorType:
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+		return oc.handleLocalPodSelectorDelFunc(
+			extraParameters.policy,
+			extraParameters.np,
+			extraParameters.portGroupIngressDenyName,
+			extraParameters.portGroupEgressDenyName,
+			obj)
+
+	case factory.LocalPodSelectorTypeVar:
+		extraParameters, ok := objectsToRetry.extraParameters.(*NetworkPolicyExtraParameters)
+		if !ok {
+			return fmt.Errorf("could not cast extraParameters to " +
+				" *NetworkPolicyExtraParameters")
+		}
+
+		return oc.handleLocalPodSelectorDelFunc(
+			extraParameters.policy,
+			extraParameters.np,
+			extraParameters.portGroupIngressDenyName,
+			extraParameters.portGroupEgressDenyName,
+			obj)
+
+	default:
+		return fmt.Errorf("object type %v not supported", objectsToRetry.oType)
+	}
+
+	return nil
+}
+
+// iterateRetryResources checks if any outstanding resource objects exist and if so it tries to
+// re-add them. updateAll forces all objects to be attempted to be retried regardless.
+func (oc *Controller) iterateRetryResources(r *retryObjs, updateAll bool) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	now := time.Now()
+	for objKey, entry := range r.entries {
+		if entry.ignore {
+			continue
+		}
+		// check if we need to create the resource object
+		if entry.newObj != nil {
+			// get the latest version of the resource object from the informer;
+			// if it doesn't exist we are not going to create the new object.
+			obj, err := oc.getResourceFromInformerCache(r.oType, objKey)
+			if err != nil && kerrors.IsNotFound(err) {
+				klog.Infof("%v %s not found in the informers cache,"+
+					" not going to retry object create", r.oType, objKey)
+				// QUESTION: shouldn't we actually delete the object?
+				entry.newObj = nil
+			} else { // TODO what if it's any other error?
```

Comment:
good question this is probably a bug in the current iterate pods retry logic. @tssurya wdyt?

```
		// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry creation
		if podEntry.needsAdd {
			kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
			if err != nil && errors.IsNotFound(err) {
				klog.Infof("%s pod not found in the informers cache, not going to retry pod setup", podDesc)
				podEntry.needsAdd = false
			} else {
				pod = kPod
			}
		}
```

Reply:
right, I think for the pods part; if we are unable to fetch the pod from kapi for whatever reason we are going to have bigger problems; if we really want to fix this better we have two options:
1) remove it completely from retryCache - which doesn't make sense since it could be a temp glitch.
2) retry this again in some time - we should do this probably with a continue, does that make sense?

Reply:
yeah I think we should do number 2, just retry again later until we get a valid response back

Reply:
looks like @ricky-rav went with 2 in his latest update, resolving this comment.